I can confirm that this fix error https://bugs.launchpad.net/mixxx/+bug/1804411 on Linux Mint 19 x64

But if I remove the line suggested by @daschuer in  #1959 I get the following error:
```
ValueError: list.remove(x): x not in list:
  File "/home/stefan/mixxx_develop/mixxx/SConstruct", line 101:
    duplicate=0, exports=['build'])
  File "/usr/lib/scons/SCons/Script/SConscript.py", line 614:
    return method(*args, **kw)
  File "/usr/lib/scons/SCons/Script/SConscript.py", line 551:
    return _SConscript(self.fs, *files, **subst_kw)
  File "/usr/lib/scons/SCons/Script/SConscript.py", line 256:
    call_stack[-1].globals)
  File "/home/stefan/mixxx_develop/mixxx/vamp-plugins/SConscript", line 62:
    env['LIBS'].remove('vamp-hostsdk')
```

but If I remove     ` env['LIBS'].remove('vamp-hostsdk')`  on line 62 in the file `/mixxx/vamp-plugins/SConscript` I get other errors:

```lin64_build/libmixxx.a(vamppluginadapter.o): In Funktion »mixxx::(anonymous namespace)::getPluginLoaderLocked()«:
/home/stefan/mixxx_develop/mixxx/src/analyzer/vamp/vamppluginadapter.cpp:149: Warnung: undefinierter Verweis auf »_VampHost::Vamp::HostExt::PluginLoader::getInstance()«
lin64_build/libmixxx.a(vamppluginadapter.o): In Funktion »mixxx::VampPluginAdapter::listPlugins[abi:cxx11]()«:
/home/stefan/mixxx_develop/mixxx/src/analyzer/vamp/vamppluginadapter.cpp:184: Warnung: undefinierter Verweis auf »_VampHost::Vamp::HostExt::PluginLoader::listPlugins[abi:cxx11]()«
lin64_build/libmixxx.a(vamppluginadapter.o): In Funktion »mixxx::VampPluginAdapter::composePluginKey(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)«:
/home/stefan/mixxx_develop/mixxx/src/analyzer/vamp/vamppluginadapter.cpp:198: Warnung: undefinierter Verweis auf »_VampHost::Vamp::HostExt::PluginLoader::composePluginKey(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)«
lin64_build/libmixxx.a(vamppluginadapter.o): In Funktion »mixxx::VampPluginAdapter::loadPlugin(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, float, int)«:
/home/stefan/mixxx_develop/mixxx/src/analyzer/vamp/vamppluginadapter.cpp:165: Warnung: undefinierter Verweis auf »_VampHost::Vamp::HostExt::PluginLoader::loadPlugin(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, float, int)«
lin64_build/libmixxx.a(vampanalyzer.o): In Funktion »VampAnalyzer::Init(QString, QString, int, int, bool)«:
/home/stefan/mixxx_develop/mixxx/src/analyzer/vamp/vampanalyzer.cpp:60: Warnung: undefinierter Verweis auf »_VampHost::Vamp::PluginHostAdapter::getPluginPath[abi:cxx11]()«
lin64_build/libmixxx.a(vampanalyzer.o): In Funktion »VampAnalyzer::GetInitFramesVector()«:
/home/stefan/mixxx_develop/mixxx/src/analyzer/vamp/vampanalyzer.cpp:247: Warnung: undefinierter Verweis auf »_VampHost::Vamp::RealTime::realTime2Frame(_VampHost::Vamp::RealTime const&, unsigned int)«
lin64_build/libmixxx.a(vampanalyzer.o): In Funktion »VampAnalyzer::GetEndFramesVector()«:
/home/stefan/mixxx_develop/mixxx/lib/vamp/vamp-sdk/RealTime.h:92: Warnung: undefinierter Verweis auf »_VampHost::Vamp::RealTime::RealTime(int, int)«
lin64_build/libmixxx.a(vampanalyzer.o): In Funktion »VampAnalyzer::GetEndFramesVector()«:
/home/stefan/mixxx_develop/mixxx/src/analyzer/vamp/vampanalyzer.cpp:263: Warnung: undefinierter Verweis auf »_VampHost::Vamp::RealTime::realTime2Frame(_VampHost::Vamp::RealTime const&, unsigned int)«
lin64_build/libmixxx.a(vampanalyzer.o): In Funktion »VampAnalyzer::Process(float const*, int)«:
/home/stefan/mixxx_develop/mixxx/src/analyzer/vamp/vampanalyzer.cpp:175: Warnung: undefinierter Verweis auf »_VampHost::Vamp::RealTime::frame2RealTime(long, unsigned int)«
collect2: error: ld returned 1 exit status
scons: *** [lin64_build/mixxx] Error 1
scons: building terminated because of errors.
```
so there might be more to do..

--- a/build/features.py
+++ b/build/features.py
@@ -355,14 +355,18 @@ class Vamp(Feature):
         # If there is no system vamp-hostsdk is installed or if the version
         # of the installed vamp-hostsdk is less than the bundled version,
         # then we'll directly link the bundled vamp-hostsdk
-        if not conf.CheckLib('vamp-hostsdk') or not conf.CheckForPKG('vamp-plugin-sdk', '2.7.1'):
+        # Note: We're adding -lvamp-hostsdk to LIBS in sources(), so
+        # don't add it now in order to prevent duplication.
+        if not conf.CheckLib('vamp-hostsdk', autoadd=False) or not conf.CheckForPKG('vamp-plugin-sdk', '2.7.1'):
             self.INTERNAL_LINK = True
             build.env.Append(CPPPATH=['#' + self.INTERNAL_VAMP_PATH])
 
         # Needed on Linux at least. Maybe needed elsewhere?
         if build.platform_is_linux:
             # Optionally link libdl Required for some distros.
-            conf.CheckLib(['dl', 'libdl'])
+            # Note: We are adding -ldl to LIBS in sources(), so
+            # don't add it now in order to prevent duplication.
+            conf.CheckLib(['dl', 'libdl'], autoadd=False)
 
         # FFTW3 support
         have_fftw3_h = conf.CheckHeader('fftw3.h')
@@ -383,10 +387,12 @@ class Vamp(Feature):
             env.SConscript(env.File('SConscript', vamp_dir))
 
             build.env.Append(LIBPATH=self.INTERNAL_VAMP_PATH)
-            build.env.Append(LIBS=['vamp-hostsdk'])
 
-            if build.platform_is_linux:
-                build.env.Append(LIBS=['dl'])
+        build.env.Append(LIBS=['vamp-hostsdk'])
+
+        # Ubuntu requires -ldl to be specified after -lvamp-hostsdk.
+        if build.platform_is_linux:
+            build.env.Append(LIBS=['dl'])
 
         return ['src/analyzer/vamp/vampanalyzer.cpp',
                 'src/analyzer/vamp/vamppluginadapter.cpp',

--- a/build/features.py
+++ b/build/features.py
@@ -355,7 +355,7 @@ class Vamp(Feature):
         # If there is no system vamp-hostsdk is installed or if the version
         # of the installed vamp-hostsdk is less than the bundled version,
         # then we'll directly link the bundled vamp-hostsdk
-        # Note: We're adding -lvamp-hostsdk to LIBS in sources(), so
+        # Note: We're adding vamp-hostsdk to LIBS in sources(), so
         # don't add it now in order to prevent duplication.
         if not conf.CheckLib('vamp-hostsdk', autoadd=False) or not conf.CheckForPKG('vamp-plugin-sdk', '2.7.1'):
             self.INTERNAL_LINK = True
@@ -364,8 +364,8 @@ class Vamp(Feature):
         # Needed on Linux at least. Maybe needed elsewhere?
         if build.platform_is_linux:
             # Optionally link libdl Required for some distros.
-            # Note: We are adding -ldl to LIBS in sources(), so
-            # don't add it now in order to prevent duplication.
+            # Note: We can't add dl to LIBS here because it needs to be added after vamp-hostsdk.
+            # See: https://bugs.launchpad.net/mixxx/+bug/1804411
             conf.CheckLib(['dl', 'libdl'], autoadd=False)
 
         # FFTW3 support
@@ -388,9 +388,10 @@ class Vamp(Feature):
 
             build.env.Append(LIBPATH=self.INTERNAL_VAMP_PATH)
 
+        # Add this here unconditionally because we're not adding it in configure().
         build.env.Append(LIBS=['vamp-hostsdk'])
 
-        # Ubuntu requires -ldl to be specified after -lvamp-hostsdk.
+        # Ubuntu requires dl to be specified after vamp-hostsdk.
         if build.platform_is_linux:
             build.env.Append(LIBS=['dl'])
 

